internal/controller/replication.storage/status.go

```diff
@@ -46,26 +49,30 @@ func setPromotedCondition(conditions *[]metav1.Condition, observedGeneration int
 }
 
 // sets conditions when volume promotion was failed.
-func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64, err string) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            err,
 		Type:               v1alpha1.ConditionCompleted,
 		Reason:             v1alpha1.FailedToPromote,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            "error detected while promotion",
```

Comment:
It is strange that we set the standard message for this condition, but custom message for the previous condition. Based on the function name this should affect only the PromotedCondition.